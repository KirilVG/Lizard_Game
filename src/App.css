.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

input {
  align-self: center;
  padding: 12px 20px;
  margin: 8px 0;
  box-sizing: border-box;
  border: none;
  border-bottom: 2px solid black;
}

.content{
  border: 1px solid gray;
  margin-top:180px;
  margin-left:380px;
  background-color: #464749;
  width:333px;
  height: 352px;
  padding: 58px 76px 0 76px;
  font: 12px Arial, Helvetica, sans-serif;
}

.nameField{
  display:inline-block;
  width:80px;
  text-align: right;
  padding: 14px 10px 0 0;
  margin:0 0 7px 0;
}
.inputField{
  display:inline-block;
  width:130px;
  margin:0;
  margin:0 0 7px 0;
}
.buttonStart{
  background-color: black;
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  border-radius: 5%;
}

.playButton{
  height: 7vh;
  width: 20vh;
  font-Size: 4vh;
}

.glassCard{
  backdrop-filter: blur(15px);
  background-Color: rgba(255,255,255,.2) !important;
  border-radius: 3em !important;
  border: 2px solid rgba(255,255,255,.1);
  background-clip: padding-box;
  box-shadow: 10px 10px 10px rgba(46, 54, 68, 0.03);
}
