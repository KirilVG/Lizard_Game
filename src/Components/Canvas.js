import React from "react";
import * as myConstants from "./Constants";

const Canvas = (props) =>{
    return (
        <canvas id={props.canvasID} width={window.innerWidth} height={window.innerHeight * myConstants.heightOffset} />
    );
}

export default Canvas;
