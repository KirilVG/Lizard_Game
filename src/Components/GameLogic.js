import Player from "./Player";
import LaneSeparator from "./LaneSeparator";
import VariableContainer from "./VariableContainer";
import CollidableObjectFactory from "./CollidableObjectFactory";
import Lane from "./Lane";
import * as myConstants from "./Constants";

class GameLogic{
    constructor(width, height, lanesNum, renderer, gameEndHandler){
        this.width = width;

        this.height = height;

        this.gameIsOver = false;

        this.renderer = renderer;

        this.lanesNum = lanesNum;

        this.valueM = myConstants.initialValueMultiplier;

        this.info = VariableContainer(
            this.width,
            this.height,
            this.lanesNum
        );

        this.speed = this.info.initialSpeed;

        this.speedUps = 0;

        this.dayNightCycles = 0;

        this.obstacleSpawnTime =
            myConstants.initialObstacleSpawnTime / this.lanesNum;

        this.consumableSpawnTime =
            myConstants.initialConsumableSpawnTime / this.lanesNum;

        this.backgroundObjects = [];

        this.groundObstacleObjects = [];

        this.aerialObstacleObjects = [];

        this.consumableObjects = [];

        this.player = new Player(
            this.info.playerOriginX,
            this.info.playerOriginY,
            this.info.playerIMGWidth,
            this.info.playerIMGHeight,
            this.info.playerHitBoxWidth,
            this.info.playerHitBoxHeight,
            this.info.playerStartingLane,
            this.lanesNum,
            this.info.laneWidth,
            myConstants.undergroundAnimationTime,
            myConstants.inAirAnimationTime,
            myConstants.jumpHeightScale,
            gameEndHandler,
        );

        for (let i = 0; i < this.lanesNum; i++) {
            let lane = new Lane(
                this.info.laneOriginX + i * this.info.laneWidth,
                this.info.laneOriginY,
                this.info.laneWidth,
                this.info.laneHeight
            );
            this.backgroundObjects.push(lane);
        }

        for (let i = 0; i <= this.lanesNum; i++) {
            let laneSep = new LaneSeparator(
                this.info.laneOriginX + i * this.info.laneWidth,
                this.info.laneOriginY,
                this.info.laneHeight
            );
            this.backgroundObjects.push(laneSep);
        }

        this.collidableObstacleObjectCreator = setInterval(
            () => this.createApproachingObstacleObject(),
            this.obstacleSpawnTime
        );
        this.animate();

        this.consumableObjectCreator = setInterval(
            () => this.createConsumableObject(),
            this.consumableSpawnTime
        );

        myConstants.setDiscoModeRemainder(0);

        myConstants.setDayNightCycle();

        this.animate();

        window.addEventListener("keydown", (e) => this.WindowKeyDown(e));
    }

    WindowKeyDown(e) {
        this.player.handleMovement(myConstants.directions[e.code]);
    }

    handleGameObjectsChanges(gameObjectsArr) {
        for (let i = 0; i < gameObjectsArr.length; i++) {
            gameObjectsArr[i].update(this.renderer, this.speed);
    
            gameObjectsArr[i].handleCollision(this.player);
    
            if (gameObjectsArr[i].terminate()) {
                gameObjectsArr.splice(i, 1);
                i--;
            }
        }
    }

    animate() {
        if (!this.gameIsOver) {
          requestAnimationFrame(() => this.animate());
    
          myConstants.setDiscoModeRemainder(myConstants.discoModeRemainder-1*this.player.valueMultiplier);
    
          //check if fuel has ended
          if (this.player.fuel <= 0) {
            this.player.handleDeath("Fuel ended");
          }
    
          //clear canvas
          this.renderer.clearScreen(this.width,this.height);
    
          //display background objects
          for (let i = 0; i < this.backgroundObjects.length; i++) {
            this.backgroundObjects[i].update(this.renderer);
          }
    
          //display game objects
          this.handleGameObjectsChanges(this.groundObstacleObjects);
    
          this.player.update(this.renderer);
    
          this.handleGameObjectsChanges(this.consumableObjects);
    
          this.handleGameObjectsChanges(this.aerialObstacleObjects);
    
          //display score
          this.renderer.displayScoreAndFuel(this.info.laneOriginX,this.info.laneOriginY,this.info.laneWidth,this.lanesNum,this.player.score,this.player.fuel)
    
          //handle speed increase
          if (
            Math.floor(
              this.player.score / myConstants.pointsNeededToIncreaseScore
            ) > this.speedUps
          )
            this.speedUp();
    
            if (
              Math.floor(
                this.player.score / myConstants.pointsNeededToChangeDayNightCycle
              ) > this.dayNightCycles
            )
              {
                this.dayNightCycles++;
                myConstants.switchDayNightCycle();
              }
    
          this.gameIsOver = this.player.isDead;
        } else {
          clearInterval(this.collidableObstacleObjectCreator);
          clearInterval(this.consumableObjectCreator);
        }
      }

    
  speedUp() {
    let prevSpeed = this.speed;
    this.speed *= myConstants.valueMultiplier;

    this.player.valueMultiplier *= myConstants.valueMultiplier;

    let diff = prevSpeed / this.speed;

    this.player.inAirAnimationTime *= diff;
    this.player.underGroundAnimationTime *= diff;

    this.obstacleSpawnTime *= diff;
    this.consumableSpawnTime *= diff;

    clearInterval(this.collidableObstacleObjectCreator);
    clearInterval(this.consumableObjectCreator);

    this.collidableObstacleObjectCreator = setInterval(
      () => this.createApproachingObstacleObject(),
      this.obstacleSpawnTime
    );

    this.consumableObjectCreator = setInterval(
      () => this.createConsumableObject(),
      this.consumableSpawnTime
    );

    this.speedUps++;
  }

  createApproachingObstacleObject() {
    let res = Math.floor(Math.random() * 3);

    switch (res) {
      case 0:
        this.aerialObstacleObjects.push(
          CollidableObjectFactory.createBird(this.info, this.lanesNum)
        );
        break;
      case 1:
        this.groundObstacleObjects.push(
          CollidableObjectFactory.createCactus(this.info, this.lanesNum)
        );
        break;
      case 2:
        this.groundObstacleObjects.push(
          CollidableObjectFactory.createSmallCactus(
            this.info,
            this.lanesNum
          )
        );
        break;
      default:
        break;
    }
  }

  createConsumableObject() {
    let consumable;

    let res = Math.floor(Math.random() * 100);

    if(res<=33) {
      consumable=CollidableObjectFactory.createDiscoConsumable(this.info, this.lanesNum)
    } else {
      consumable=CollidableObjectFactory.createWorm(this.info, this.lanesNum)
    }

    this.consumableObjects.push(consumable);
  }
}

export default GameLogic;
/* displayScoreAndFuel() {
        //set the background
        this.DiscoFill(myConstants.primaryColor);
    
        this.ctx.fillRect(
          this.info.laneOriginX,
          this.info.laneOriginY,
          this.lanesNum * this.info.laneWidth,
          this.info.laneWidth * 2
        );

        //set the score
    this.DiscoFill(myConstants.secondaryColor);
    let fontSize = this.info.laneWidth * myConstants.fontScale;
    this.ctx.font = `${fontSize}px serif`;
    this.ctx.fillText(
      `score:${Math.round(this.player.score)}`,
      this.info.laneOriginX,
      this.info.laneOriginY + fontSize
    );

    //set the fuel bar
    this.DiscoFill(myConstants.secondaryColor);
    this.ctx.fillRect(
      this.info.laneOriginX + 0.1 * this.lanesNum * this.info.laneWidth,
      this.info.laneOriginY + 0.9 * this.info.laneWidth,
      (this.player.fuel / myConstants.maxFuel) *
        (this.lanesNum * this.info.laneWidth * 0.8),
      this.info.laneWidth
    );
  }*/