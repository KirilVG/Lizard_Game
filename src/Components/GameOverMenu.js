import React from "react";
import { useState } from "react";
import * as myConstants from "./Constants";
import "bootstrap/dist/css/bootstrap.min.css";
import { createTheme } from "@mui/material/styles";
import { ThemeProvider } from "@mui/material";
import { Stack } from "react-bootstrap";
import FormControl from "@mui/material/FormControl";

function GameOverMenu(props) {
  const [gameScoreSaved, setGameScoreSaved] = useState(false);

  const clickContinue = () => {
    props.onClick();
  };

  const saveScore = () => {
    let lanesAsString = String(props.end.lanesNum);

    let scoreboard = JSON.parse(localStorage.getItem("scoreboard"));
    if (!scoreboard) scoreboard = {};

    let currentScores = scoreboard[lanesAsString];
    if (!currentScores) currentScores = [];

    currentScores.push({
      username: props.end.username,
      lanesNum: props.end.lanesNum,
      score: props.end.score,
      causeOfDeath: props.end.message,
      discMode: myConstants.useDiscoMode,
    });

    currentScores.sort((a, b) => b.score - a.score);

    if (currentScores.length > myConstants.maximumNumberOfScores)
      currentScores.pop();

    scoreboard[lanesAsString] = currentScores;

    localStorage.setItem("scoreboard", JSON.stringify(scoreboard));

    setGameScoreSaved(true);
  };

  const displayScores = () => {
    let displayValue;
    let lanesAsString = String(props.end.lanesNum);

    let scoreboard = JSON.parse(localStorage.getItem("scoreboard"));

    if (!scoreboard || !scoreboard[lanesAsString]) {
      displayValue = (
        <div>
          <h1>No current scores!</h1>
          <p>be the first</p>
        </div>
      );
    } else {
      displayValue = [];

      for (
        let i = 0;
        i < scoreboard[lanesAsString].length &&
        i < myConstants.maximumNumberOfScores;
        i++
      ) {
        displayValue.push(
          <li key={i}>
            <div className="scoreItem">
              <h1 className="scoreItemHeader">{`${scoreboard[lanesAsString][i].username}`}</h1>
              <p className="scoreItemP">
                {`score: ${scoreboard[lanesAsString][i].score}, cause of death:${scoreboard[lanesAsString][i].causeOfDeath}`}
              </p>
            </div>
          </li>
        );
      }
    }

    return displayValue;
  };

  return (
    <div
      className="bg-image d-flex justify-content-center align-items-center backgroundIMG"
      style={{
        backgroundImage: `url("${myConstants.gameOverMenuBackgroundIMGUrl}")`,
      }}
    >
      <div className="mask darkMask">
        <div className="justify-content-center align-items-center h-100 container">
          <div className="row d-flex justify-content-center cardHolder">
            <div className="col-12 col-md-10 col-lg-7 col-xl-6">
              <div className="card glassCard card-body p-3 text-white my-4">
                <h2 className="text-center mb-5">Game Over!</h2>
                <Stack spacing={3} class="formStack">
                  <div>
                    <label>{`Username:${props.end.username}, cause of death:${props.end.message}, score: ${props.end.score}`}</label>
                  </div>
                  {!gameScoreSaved ? (
                    <div>
                      <button
                        type="button"
                        className="btn btn-light playButton"
                        onClick={saveScore}
                      >
                        save Score
                      </button>
                    </div>
                  ) : null}
                  <div>
                    <button
                      type="button"
                      className="btn btn-light playButton"
                      onClick={clickContinue}
                    >
                      Play Again
                    </button>
                  </div>
                </Stack>
              </div>
            </div>
            <div className="col-12 col-md-10 col-lg-7 col-xl-6">
              <div className="card glassCard card-body p-1 text-white my-4">
                <h2 className="text-center mb-5">Scores</h2>
                <Stack spacing={5} className="formStack">
                  <ol className="scoreArea">{displayScores()}</ol>
                </Stack>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default GameOverMenu;