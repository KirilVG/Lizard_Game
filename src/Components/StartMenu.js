import React from "react";
import * as myConstants from "./Constants";

function StartMenu(props) {
  const [inputText, setInputText] = React.useState("3");
  const [inputUsername, setInputUsername] = React.useState("Noname");

  const handleInput = (event) => {
    setInputText(event.target.value);
  };

  const handleUsernameInput = (event) => {
    setInputUsername(event.target.value);
  };

  const clickStart = () => {
    let lanesNum = Number(inputText);
    if (!lanesNum) {
      alert("lanes should be a number");
    } else if (lanesNum % 1 !== 0) {
      alert("lanes should be a whole number");
    } else if (lanesNum < myConstants.minLanesNum) {
      alert(`lanes should not be less than ${myConstants.minLanesNum}`);
    } else if (lanesNum > myConstants.maxLanesNum) {
      alert(`lanes should not be more than ${myConstants.maxLanesNum}`);
    } else {
      props.onClick({lanesNum: lanesNum, username:inputUsername});
    }
  };

  return (
    <div>
      <h1>Welcome to Run Kiro the Lizard.</h1>
      <label>
        Lanes:
        <input onChange={handleInput} placeholder="3" />
      </label>
      <label>
        Username:
        <input onChange={handleUsernameInput} placeholder="Noname" />
      </label>
      <button onClick={clickStart}>Start</button>
    </div>
  );
}

export default StartMenu;
