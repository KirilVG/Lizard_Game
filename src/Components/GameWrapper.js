import React, { useEffect } from "react";
import Canvas from "./Canvas";
import GameLogic from "./GameLogic";
import Canvas2DRenderer from "./Canvas2DRenderer";

const GameWrapper = (props) => {
  const { id } = props;

  useEffect(() => {
    const gameCanvas = document.getElementById(id);

    const canvas2dRenderer = new Canvas2DRenderer(gameCanvas);

    const gameLogic = new GameLogic(
      gameCanvas.width,
      gameCanvas.height,
      props.lanesNum,
      canvas2dRenderer,
      props.gameEndHandler
    );
  }, []);

  return (
    <div>
      <Canvas canvasID={id}></Canvas>
    </div>
  );
};

export default GameWrapper;
