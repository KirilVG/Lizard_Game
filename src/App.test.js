import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import StartMenu from './Components/StartMenu';
import Canvas from './Components/Canvas';
import GameOverMenu from './Components/GameOverMenu';
import GameWrapper from './Components/GameWrapper';
import GameLogic from './Components/GameLogic';
import 'jest-canvas-mock';

test('renders title element', () => {
  render(<App />);
  const linkElement = screen.getByText(/Welcome to run Kiro the lizard!/i);

  expect(linkElement).toBeInTheDocument();
});

test('renders lanes input', () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);

  expect(lanesInput).toBeInTheDocument();
});

test('renders username input', () => {
  render(<App />);
  const usernameInput = screen.getByLabelText(/Username/i);

  expect(usernameInput).toBeInTheDocument();
});

test('button should be rendered', () => {
  render(<App/>);
  const buttonEl=screen.getByRole("button");

  expect(buttonEl).toBeInTheDocument();
});

test('lanes input should change', () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue="2";

  fireEvent.change(lanesInput,{target:{value:testValue}});

  expect(lanesInput.value).toBe(testValue);
});

test('lanes input should set error prop to true when play button is pressed and Lanes input does not contain a number', () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue="a";
  const buttonEl=screen.getByRole("button");

  fireEvent.change(lanesInput,{target:{value:testValue}});
  fireEvent.click(buttonEl);

  const linkElement = screen.getByText(/lanes should be a number/i);

  expect(linkElement).toBeInTheDocument();
});

test('lanes input should set error prop to true when play button is pressed and Lanes input contains a number less than the minimum', () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue="2";
  const buttonEl=screen.getByRole("button");

  fireEvent.change(lanesInput,{target:{value:testValue}});
  fireEvent.click(buttonEl);

  const linkElement = screen.getByText(/lanes should not be less than 3/i);

  expect(linkElement).toBeInTheDocument();
});

test('lanes input should set error prop to true when play button is pressed and Lanes input contains a number more than the maximum', () => {
  render(<App />);
  const lanesInput = screen.getByLabelText(/Number of lanes/i);
  const testValue="102";
  const buttonEl=screen.getByRole("button");

  fireEvent.change(lanesInput,{target:{value:testValue}});
  fireEvent.click(buttonEl);

  const linkElement = screen.getByText(/lanes should not be more than 100/i);

  expect(linkElement).toBeInTheDocument();
});

test('username input should change', () => {
  render(<App />);
  const usernameInput = screen.getByLabelText(/Username/i);
  const testValue="User";

  fireEvent.change(usernameInput,{target:{value:testValue}});

  expect(usernameInput.value).toBe(testValue);
});

it('game start renders correctly', () => {
  const lanesNum=null;
  const username=null;
  const onClick = () => {};
  const tree = renderer
    .create(<StartMenu onClick={onClick} lanesNum={lanesNum} username={username} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('game end renders correctly', () => {
  const gameOverInf={
    score: 100,
    cause: "killed by sth",
    username: "Person",
    lanesNum: 3,
  };
  const onClick = () => {};
  const tree = renderer
    .create(<GameOverMenu onClick={onClick} end={gameOverInf}  />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render canvas', () => {
  const demoId="demoId";
  const tree = renderer
    .create(<Canvas canvasID={demoId} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render game wrapper', () => {
  const demoId = "demoId";
  const lanesNum = 3;
  const handleGameEnd=()=>{};
  const tree = renderer
    .create(<GameWrapper id={demoId} lanesNum={lanesNum} gameEndHandler={handleGameEnd} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("game logic executes", () => {
  const demoWidth = 1000;
  const demoHeight = 1000;
  const demoId = "demoId";
  const lanesNum = 3;
  const handleGameEnd = () => {};
  const canvas = document.createElement("canvas");

  //const gameLogic = new GameLogic(canvas,lanesNum,handleGameEnd);

  expect(canvas.getContext("2d")).not.toBe(undefined);
});




