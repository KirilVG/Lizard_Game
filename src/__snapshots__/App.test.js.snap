// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game end renders correctly 1`] = `
<div
  className="bg-image d-flex justify-content-center align-items-center backgroundIMG"
  style={
    Object {
      "backgroundImage": "url(\\"https://images.unsplash.com/photo-1573595015625-ac674ed1d155?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1000&q=80\\")",
    }
  }
>
  <div
    className="mask darkMask"
  >
    <div
      className="justify-content-center align-items-center h-100 container"
    >
      <div
        className="row d-flex justify-content-center cardHolder"
      >
        <div
          className="col-12 col-md-10 col-lg-7 col-xl-6"
        >
          <div
            className="card glassCard card-body p-3 text-white my-4"
          >
            <h2
              className="text-center mb-5"
            >
              Game Over!
            </h2>
            <div
              class="formStack"
              className="vstack minBreakpoint-xs"
              spacing={3}
            >
              <div>
                <label>
                  Username:Person, cause of death:undefined, score: 100
                </label>
              </div>
              <div>
                <button
                  className="btn btn-light playButton"
                  onClick={[Function]}
                  type="button"
                >
                  save Score
                </button>
              </div>
              <div>
                <button
                  className="btn btn-light playButton"
                  onClick={[Function]}
                  type="button"
                >
                  Play Again
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-10 col-lg-7 col-xl-6"
        >
          <div
            className="card glassCard card-body p-1 text-white my-4"
          >
            <h2
              className="text-center mb-5"
            >
              Scores
            </h2>
            <div
              className="formStack vstack minBreakpoint-xs"
              spacing={5}
            >
              <ol
                className="scoreArea"
              >
                <div>
                  <h1>
                    No current scores!
                  </h1>
                  <p>
                    be the first
                  </p>
                </div>
              </ol>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render canvas 1`] = `
<canvas
  height={760.3199999999999}
  id="demoId"
  width={1024}
/>
`;

exports[`should render game wrapper 1`] = `
<div>
  <canvas
    height={760.3199999999999}
    id="demoId"
    width={1024}
  />
</div>
`;

exports[`start menu renders correctly 1`] = `
<div
  className="bg-immage d-flex justify-content-center align-items-center backgroundIMG"
  style={
    Object {
      "backgroundImage": "url(\\"https://live.staticflickr.com/65535/48701594348_e099758f0c_b.jpg\\")",
    }
  }
>
  <div
    className="mask darkMask d-flex justify-content-center align-items-center h-100"
  >
    <div
      className="container row justify-content-center"
    >
      <div
        className="col-12 col-md-10 col-lg-7 col-xl-6"
      >
        <div
          className="MuiFormControl-root card glassCard card-body inputForm css-1nrlq1o-MuiFormControl-root"
        >
          <h2
            className="text-center mb-5 formItem"
          >
            Welcome to run Kiro the lizard!
          </h2>
          <div
            className="formStack css-pnq1z9-MuiStack-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root css-19lnug-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor="outlined-basic"
                id="outlined-basic-label"
              >
                Number of lanes
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  defaultValue={null}
                  disabled={false}
                  id="outlined-basic"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Number of lanes
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root css-19lnug-MuiFormControl-root-MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={false}
                htmlFor="outlined-basic"
                id="outlined-basic-label"
              >
                Username
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  defaultValue={null}
                  disabled={false}
                  id="outlined-basic"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-yjsfm1"
                  >
                    <span>
                      Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              className="btn btn-light playButton"
              onClick={[Function]}
              type="button"
            >
              Play
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
